### PR TITLE
Implement support for the GL_LINE_LOOP primitive

### PR DESCRIPTION
**Pull request by bythos14**

### CODE REVIEW

File: source/ffp.c [hunk 1]
```diff
@@ -1071,17 +1071,25 @@ void glEnd(void) {
 	// Restoring original attributes state settings
 	ffp_vertex_attrib_state = orig_state;
 
-	if (prim_is_non_native)
-		if (prim == SCE_GXM_PRIMITIVE_TRIANGLES) // GL_QUADS
+	if (prim_is_non_native) {
+		if (ffp_mode == GL_QUADS)
```
Comment (Rinnegatamante) [inline source/ffp.c:1075]: With 3 if cases here i think it's time to switch to a switch statement

File: source/vitaGL.c [hunk 2]
```diff
@@ -923,25 +923,33 @@ void glDrawArrays(GLenum mode, GLint first, GLsizei count) {
...
-	
+
 #ifndef SKIP_ERROR_HANDLING
 	if (is_draw_legal)
 #endif
 	{
-		if (prim_is_non_native)
-			if (gxm_p == SCE_GXM_PRIMITIVE_TRIANGLES) // GL_QUADS
+		if (prim_is_non_native) {
+			if (mode == GL_QUADS) // GL_QUADS
```
Comment (Rinnegatamante) [inline source/vitaGL.c:940]: switch statement ditto

File: source/vitaGL.c [hunk 3]
```diff
@@ -976,77 +984,57 @@ void glDrawElements(GLenum mode, GLsizei count, GLenum type, const GLvoid *gl_in
...
+		// Directly use the current IBO if there is no need for a temporary index buffer.
+		if (gpu_buf != NULL && !prim_is_non_native) {
+			ptr = (uint16_t *)((uint8_t *)gpu_buf->ptr + (uint32_t)gl_indices);
+			gpu_buf->used = GL_TRUE;
+		} else {
+			int i;
+			uint16_t *dst;
+			uint16_t *src = gpu_buf != NULL ? (uint16_t *)((uint8_t *)gpu_buf->ptr + (uint32_t)gl_indices) : (uint16_t *)gl_indices;
+
+			if (mode == GL_QUADS) {
```
Comment (Rinnegatamante) [inline source/vitaGL.c:997]: switch ditto

**Review by Rinnegatamante — CHANGES_REQUESTED** [review]

**Comment by bythos14** [comment]
This good?
405dd95

**Pull request merged**